v2/datastream-to-spanner/terraform/samples/mysql-sharded-end-to-end/README.md

```diff
@@ -0,0 +1,689 @@
+of this is done seamlessly by Dataflow and there is no risk to the fidelity of
+an already executing job.
+
+Example update: Changing `round_json_decimals` to `true` from `false`.
+
+Look for the following log during `terraform apply` -
+
+```shell
+  # google_dataflow_flex_template_job.live_migration_job will be updated in-place
+```
+
+### Updating workers of a Dataflow job
+
+Currently, the Terraform `google_dataflow_flex_template_job` resource does not
+support updating the workers of a Dataflow job.
+If the worker counts are changed in `tfvars` and a Terraform apply is run,
+Terraform will attempt to cancel/drain the existing Dataflow job and replace it
+with a new one.
+**This is not recommended**. Instead use the `gcloud` CLI to update the worker
+counts of a launched Dataflow job.
+
+```shell
+gcloud dataflow jobs update-options \                                                                                                                                                                                                                                                                                                                      (base) 
+        --region=us-central1 \
+        --min-num-workers=5 \
+        --max-num-workers=20 \
+      2024-06-17_01_21_44-12198433486526363702
+```
+
+### Long time to delete SMT bucket
+
+A GCS bucket can only be deleted if all its comprising objects are deleted
+first. This template attempts to delete all objects in the created GCS bucket
+and then deletes the bucket as well.
+**This may take a long time, depending on the size of the data in the bucket.**
+If you want Terraform to only create the GCS bucket but skip its deletion
+during `terraform destroy`, you will have to use the `terraform state rm` API to
+delete GCS resource from being traced by Terraform after the `apply` command.
+
+### Configuring Databases and Tables in Datastream
+
+Which databases and tables to replicate can be configured via the following
+variable definition -
+
+In `variables.tf`, following definition exists -
+
+```shell
+mysql_databases = list(object({
+      database = string
+      tables   = optional(list(string))
+    }))
+```
+
+To configure, create `*.tfvars` as follows -
+
+```shell
+mysql_databases = [
+    {
+      database = "<YOUR_DATABASE_NAME>"
+      tables   = ["TABLE_1", "TABLE_2"]
+      # Optionally list specific tables, or remove "tables" all together for all tables
+    },
+    {
+      database = "<YOUR_DATABASE_NAME>"
+      tables   = ["TABLE_1", "TABLE_2"]
+      # Optionally list specific tables, or remove "tables" all together for all tables
+    },
+  ]
+```
+
+### Specifying schema overrides
+
+By default, the Dataflow job performs a like-like mapping between
+source and Spanner. Any schema changes between source and Spanner can be
+specified using the `session file`. To specify a session file -
+
+1. Copy the
+   contents of the SMT generated `session file` to the `session.json` file.
```

Comment:
You should also be able to point to an existing file right ? May be that should be the suggested approach rather than asking folks to copy, which may not work if the file is too big.

Reply:
Yes, this is what the code does - I think I made the instructions a bit confusing to understand. Rephrasing to make it clearer.